third_party/move/tools/move-lint/src/lint/rules/bool_comparison.rs

```diff
@@ -0,0 +1,157 @@
+//! Detects comparisons where a variable is compared to 'true' or 'false' using
+//! equality (==) or inequality (!=) operators and provides suggestions to simplify the comparisons.
+//! Examples: if (x == true) can be simplified to if (x), if (x == false) can be simplified to if (!x)
+use crate::lint::utils::{add_diagnostic_and_emit, get_var_info_from_func_param, LintConfig};
+use crate::lint::visitor::ExpressionAnalysisVisitor;
+
+use codespan::FileId;
+use codespan_reporting::diagnostic::Diagnostic;
+use move_model::ast::{ExpData, Operation, Value};
+use move_model::model::{FunctionEnv, GlobalEnv};
+pub struct BoolComparisonVisitor;
+
+impl Default for BoolComparisonVisitor {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl BoolComparisonVisitor {
+    pub fn new() -> Self {
+        Self {}
+    }
+    pub fn visitor() -> Box<dyn ExpressionAnalysisVisitor> {
+        Box::new(Self::new())
+    }
+
+    // Get the variable name or function name from a given expression. This will be used to
+    // print out the message for this lint.
+    fn get_var_name_or_func_name_from_exp(
+        &mut self,
+        exp: &ExpData,
+        func_env: &FunctionEnv,
+        env: &GlobalEnv,
+    ) -> Option<String> {
+        match exp {
+            ExpData::Temporary(_, index) => {
+                let parameters = func_env.get_parameters();
+                let param = get_var_info_from_func_param(*index, &parameters).unwrap();
```

Comment:
Please do a grep on all `unwrap`. Replace them with an `expect` if it can be guaranteed to expect safely, if not make sure we do not panic and report a user friendly error.

Reply:
refactored